mesonbuild/utils/universal.py

```diff
@@ -184,6 +186,11 @@ class _VerPickleLoadable(Protocol):
     python_command = [sys.executable]
 _meson_command: T.Optional['ImmutableListProtocol[str]'] = None
 
+if sys.version_info >= (3, 11):
+    getencoding = locale.getencoding
+else:
+    getencoding = locale.getpreferredencoding
```

Comment:
This upstream python warning is garbage, I seem to recall I filtered it out explicitly to prevent errors in the testsuite. I don't care if it collects warnings in the testsuite. Warnings are not errors, especially when the warnings are fallaciously incorrect.